plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/Base85Codec.java

```diff
@@ -61,19 +61,18 @@ private Base85Codec() {}
 
     public static ByteBuffer decode(String encoded)
     {
-        char[] input = encoded.toCharArray();
-        checkArgument(input.length % 5 == 0, "Input should be 5 character aligned");
-        ByteBuffer buffer = ByteBuffer.allocate(input.length / 5 * 4);
+        checkArgument(encoded.length() % 5 == 0, "Input should be 5 character aligned");
+        ByteBuffer buffer = ByteBuffer.allocate(encoded.length() / 5 * 4);
 
         int inputIndex = 0;
-        while (inputIndex < input.length) {
-            int sum = 0;
-            sum += decodeInputChar(input[inputIndex]) * BASE_4TH_POWER;
-            sum += decodeInputChar(input[inputIndex + 1]) * BASE_3RD_POWER;
-            sum += decodeInputChar(input[inputIndex + 2]) * BASE_2ND_POWER;
-            sum += decodeInputChar(input[inputIndex + 3]) * BASE;
-            sum += decodeInputChar(input[inputIndex + 4]);
-            buffer.putInt(sum);
+        while (inputIndex < encoded.length()) {
+            long sum = 0;
+            sum += decodeInputChar(encoded.charAt(inputIndex)) * BASE_4TH_POWER;
+            sum += decodeInputChar(encoded.charAt(inputIndex + 1)) * BASE_3RD_POWER;
+            sum += decodeInputChar(encoded.charAt(inputIndex + 2)) * BASE_2ND_POWER;
+            sum += decodeInputChar(encoded.charAt(inputIndex + 3)) * BASE;
+            sum += decodeInputChar(encoded.charAt(inputIndex + 4));
+            buffer.putInt((int) sum); // This causes overflow, but it's same as Delta Lake
```

Comment:
this sounds wrong and requires further investigation

Reply:
I don't think it's "wrong". The expected behavior is doing the same behavior as Delta Lake.

Reply:
I don't see how decoding logic could legitimately overflow. That's why i think that this requires investigation.

Reply:
I think i understand now.
The int is a container for 32 bits (including the sign bit), whereas `toIntExact` works only when 31 bits are set.

Let's switch `sum` back to `int` and add a comment.

```
// word is a container for 32-bit of decoded output. Arithmetics below treat it as unsigned 32-bit integer (consciously overflow)
int word = 0;
word += decodeInputChar(encoded.charAt(inputIndex)) * BASE_4TH_POWER;
word += decodeInputChar(encoded.charAt(inputIndex + 1)) * BASE_3RD_POWER;
word += decodeInputChar(encoded.charAt(inputIndex + 2)) * BASE_2ND_POWER;
word += decodeInputChar(encoded.charAt(inputIndex + 3)) * BASE;
word += decodeInputChar(encoded.charAt(inputIndex + 4));
buffer.putInt(word);
```

Reply:
Alternatively we can keep the variable as `long` (and avoid any overflows during arithemtics)
and then just check it has at most 32 bits set before casting to int.